cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/timeseries/FactTable.java

```diff
@@ -121,11 +125,79 @@ public FactScanner scan(FactScan scan) throws Exception {
                              scan.getStartTs(), scan.getEndTs());
   }
 
+  /**
+   * Given a list of previous tags, we return a list of available tags at the child context in the provided time range.
```

Comment:
Why do you refer to those as previous tags. Not clear "previous" from where? How about:

Given the list of tag values, return collection of tagValues that appear next to the given tagValues (on first position right after given ones) in any of the written facts in specified time range.